sdk/core/azure-core/azure/core/polling/_poller.py

```diff
@@ -240,7 +255,7 @@ def wait(self, timeout: Optional[float] = None) -> None:
          operation to complete (in seconds).
         :raises ~azure.core.exceptions.HttpResponseError: Server problem with the query.
         """
-        if self._thread is None:
+        if not self._is_alive():
```

Comment:
This is changing the behaviour, as it will skip the raised exception in the case that the polling thread has already finished. Will fix.

Reply:
If the initial call was terminal (which happens at time), the thread was never started. It seems now it will start, I think we should keep the behavior to never start it if the initial call is terminal.

Reply:
If the initial call was terminal - will the `self._thread` be set? Or will it be `None`?
This change will only start a thread that was created in the constructor, so if it wasn't created there it wont be started.